### PR TITLE
Pz/CUSTOMERS API - list activities

### PR DESCRIPTION
**Pull request by p-zielinski**
## 20231107 - Customers API

https://docs.voucherify.io/v2018-08-01-piotr-542/reference/customer-activity-object

**Added schemas**

- SimpleOrder
- SimpleOrderItem
- SimplePromotionTier
- SimpleRedemption
- ReferralProgram
- SimpleVoucher
- SimpleConsent
- CustomersListCustomerActivitiesResponseBody
- CustomerActivity
- CustomerActivityData
- EventCustomerSent
- EventCustomerRecovered
- EventCustomerFailed
- EventCustomerConfirmed
- EventCustomerCreated
- EventCustomerUpdated
- EventCustomerDeleted
- EventCustomerReferred
- EventCustomerCustomEvent
- EventCustomerSegmentEntered
- EventCustomerSegmentLeft
- EventCustomerSmsSent
- EventCustomerSmsRecovered
- EventCustomerSmsFailed
- EventCustomerEmailSent
- EventCustomerEmailRecovered
- EventCustomerEmailFailed
- EventCustomerActiveCampaignSent
- EventCustomerActiveCampaignRecovered
- EventCustomerActiveCampaignFailed
- EventCustomerBrazeSent
- EventCustomerBrazeRecovered
- EventCustomerMailchimpSent
- EventCustomerMailchimpRecovered
- EventCustomerMailchimpFailed
- EventCustomerIntercomSent
- EventCustomerIntercomRecovered
- EventCustomerIntercomFailed
- EventCustomerShopifySent
- EventCustomerShopifyRecovered
- EventCustomerShopifyFailed
- EventCustomerKlaviyoSent
- EventCustomerKlaviyoRecovered
- EventCustomerKlaviyoFailed
- EventCustomerBatchSent
- EventCustomerBatchRecovered
- EventCustomerBatchFailed
- EventCustomerRewarded
- EventCustomerRewardedLoyaltyPoints
- EventCustomerGiftVoucherBalanceAdded
- EventCustomerLoyaltyCardPointsAdded
- EventCustomerLoyaltyCardPointsTransferred
- EventCustomerVouchersLoyaltyPointsExpired
- EventCustomerVoucherDeleted
- EventCustomerPublicationSucceeded
- EventCustomerPublicationFailed
- EventCustomerValidationSucceeded
- EventCustomerValidationFailed
- EventCustomerRedemption
- EventCustomerRedemptionSucceeded
- EventCustomerRedemptionFailed
- EventCustomerRedemptionRollbackSucceeded
- EventCustomerRedemptionRollbackFailed
- EventCustomerConsents
- EventCustomerConsentsGiven
- EventCustomerConsentsRevoked
- EventCustomerOrder
- EventCustomerOrderCanceled
- EventCustomerOrderCreated
- EventCustomerOrderFulfilled
- EventCustomerOrderPaid
- EventCustomerOrderProcessing
- EventCustomerOrderUpdated
- EventCustomerRewardRedemptions
- EventCustomerRewardRedemptionsCreated
- EventCustomerRewardRedemptionsPending
- EventCustomerRewardRedemptionsCompleted
- EventCustomerRewardRedemptionsRolledBack
- EventCustomerLoyaltyUpdated
- EventCustomerLoyaltyTierBase
- EventCustomerLoyaltyTierUpgraded
- EventCustomerLoyaltyTierDowngraded
- EventCustomerLoyaltyTierProlonged
- EventCustomerLoyaltyTierExpirationChanged
- EventCustomerLoyaltyTierJoined
- EventCustomerLoyaltyTierLeft
- ValidationRulesListRulesAssignmentsResponseBody
- SimpleReferralTier
- CustomEvent
- ValidationEntity
- SimpleSku
- SimpleProduct
- LoyaltyTierExpiration
- VoucherTransaction
- LoyaltyCardTransaction

- GET `/v1/customers/{customerId}/activities`
  - Request body schema was replaced with new one: `CustomersListCustomerActivitiesResponseBody` (old one: `9_res_list_customer_activities`)

- **reward_assignment** renamed to **RewardAssignment**
- **validation_rules_list_rules_assignments_response_body** renamed to **ValidationRulesListRulesAssignmentsResponseBody**

### CODE REVIEW

File: reference/OpenAPI.json [hunk 1]
```diff
@@ -26765,9 +27066,241 @@
...
+            "enum": [
+              "consent"
+            ]
+          }
+        },
+        "required": [
+          "object"
+        ]
+      },
+      "CustomersListCustomerActivitiesResponseBody": {
```
Comment (jkaliszuk) [inline reference/OpenAPI.json:27301]: Tu chyba required powinno być na tych polach.

File: reference/OpenAPI.json [hunk 2]
```diff
@@ -26857,571 +27429,1364 @@
...
+          "customer": {
+            "$ref": "#/components/schemas/SimpleCustomer"
+          },
+          "distribution": {
+            "type": "object"
+          },
+          "sent_at": {
+            "type": "string",
+            "example": "2021-12-22T10:13:06.487Z",
+            "description": "Timestamp representing the date and time when the sms was sent in ISO 8601 format.",
```
Comment (jkaliszuk) [inline reference/OpenAPI.json:27658]: To chyba nie będzie tylko SMS z tego co widzę. Bo jest użyte w kilku rodzajach dystrybucji.

File: reference/OpenAPI.json [hunk 3]
```diff
@@ -26857,571 +27429,1364 @@
...
+          "customer": {
+            "$ref": "#/components/schemas/SimpleCustomer"
+          },
+          "distribution": {
+            "type": "object"
+          },
+          "recovered_at": {
+            "type": "string",
+            "example": "2021-12-22T10:13:06.487Z",
+            "description": "Timestamp representing the date and time when the sms was recovered in ISO 8601 format.",
```
Comment (jkaliszuk) [inline reference/OpenAPI.json:27679]: Tutaj chyba też nie sms

**Review by jkaliszuk — CHANGES_REQUESTED** [review]

File: reference/OpenAPI.json [hunk 4]
```diff
@@ -9086,6 +9238,86 @@
...
+          },
+          "name": {
+            "type": "string",
+            "description": "Name of the promotion tier."
+          },
+          "banner": {
+            "type": "string",
+            "description": "Text to be displayed to your customers on your website."
+          },
+          "campaign": {
```
[the remark this answers is not in the document]
Comment (p-zielinski) [reply reference/OpenAPI.json:9262]: racja

**Review by jkaliszuk — APPROVED** [review]

**Pull request merged**